engine/src/mblandroidjava.cpp

```diff
@@ -36,7 +36,11 @@ static jmethodID s_integer_constructor = nil;
 static bool init_integer_class(JNIEnv *env)
 {
     if (s_integer_class == nil)
```

Comment:
Don't check for nil in any of this init_* functions. Make sure that they are only ever called once on startup - so you can guarantee you need to fetch refs again.